ptrace.c

```diff
@@ -233,16 +233,16 @@ extern inline bool sm_peekdata(const void *addr, uint16_t length, const mem64_t
 
     /* check if we have a full cache hit */
     if (peekbuf.base != NULL &&
-        reqaddr >= peekbuf.base &&
-        (unsigned long) (reqaddr + length - peekbuf.base) <= peekbuf.size)
+        peekbuf.base - reqaddr <= 0 &&
```

Comment:
Is this line really faster? Mine gives roughly the same speed with or without this change

Reply:
The change is really really tiny, but it's consistently 1-2% faster this way.
The other line is the big one, ofc, that gives the remaining ~10%

Of course it depends on compiler and microarchitecture, your compiler may even output the same executable for both (mine doesn't).

Happy to hear it works for you as well, if the current line seems too obscure for you I can give up a 1% for clearer code.

Reply:
Yeah, I think we should keep the original line for clarity. The rest looks good to merge